src/arm7tdmi.rs

```diff
@@ -93,8 +93,14 @@ impl Arm7tdmi {
         println!("PC: {:?}", self.program_counter);
     }
 
-    fn branch_link(&mut self, _op_code: u32) {
-        todo!("Branch Link")
+    fn branch_link(&mut self, op_code: u32) {
+        self.registers[14] = self.program_counter;
+
+        let offset = op_code & 0b0000_0000_1111_1111_1111_1111_1111_1111;
+        println!("offset: {:?}", offset);
+
+        self.program_counter += 8 + offset * 4;
```

Comment:
This depends on opcode at bit 24, there are 3 different implementation, please choose one to implement, this seems `B` (=0)

Reply:
Look at this
![Immagine 2022-09-16 104912](https://user-images.githubusercontent.com/99926684/190597679-50ea1d27-58ce-49f6-a341-50d6c305a3c9.png)

`B` (=0) is `Branch` 
`BL` (=1) is `BranchLink`
as you can see in this implementation.

![Immagine 2022-09-16 110807](https://user-images.githubusercontent.com/99926684/190601387-4a029ff7-10cc-4b21-a1d0-de359ec4d4f2.png)

The only difference is that `BL` saves `PC + 4` in `LR` (R14)

We wrote `B` and `BL` separately.
Let's talk about it if we want to write them in a function.

P.N.
Exist also `BLX`
![Immagine 2022-09-16 111244](https://user-images.githubusercontent.com/99926684/190602320-11afa63d-b7b8-4e03-a167-5443b6c2b62a.png)

Reply:
Then, the only missing implementation is BLX, right?

Reply:
Yes, exactly! I have a question about this...
- What does `Opcode` = `H` mean?
I could suppose `Halfword Offset for BLX`, so we cannot implement `BLX` like `Branch` and `Branch With Link` because we cannot set 24th bit to`H` (obviously).

Reply:
I can open an issue about this implementation.

Reply:
Ok for now it's ok, during next interactions we will refactor this stuff in more suited way

Reply:
I wrote in the previous comment If we want to write 'branch' and 'branch with link' in a single function. I think we should write this in a suited way, as you said.